### PR TITLE
States

### PR DESCRIPTION
**Pull request by mtp000**
_**_gameplay.js_**_
- altered gameplay state machine to reflect that each state is a outcome scene from the storyboard
- renamed scenes to be more descriptive
- added xstate visualizer extension

_**App.jsx**_
- added dynamic mapping of states to reduce redundancy of conditional statements
All other components created in states/flow1 are scenes from storyboard for the illegal option of game

### CODE REVIEW

**Comment by mtp000** [comment]
For catalyst_earth and catalyst_country, I render these states conditionally inside the Catalyst component, whereras for other states that also only have a continue button, I render the states in their own seperate components. Would best practice be to group similar scenes like Catalysts scene or create seperate components for every state?

**Comment by mtp000** [comment]
I'm also getting error " {send} is missing in props validation" so i turned this rule off in eslintrc file.

File: phaser-react/src/gameplay.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
...
...Vi4hPIARgBWYESqAG5grPwAMtsiAOIAgts0nAAKAjw0AMoie+y8IgAikjJIIOUqaprvNXUNTVCLTaHS6PT6iFCPiM1hCeiM5gcoQMenaehicXQMySC2WqyUG1YeyOJ3Ol24NzuIgez1eZUUXyqvy81j0AQIAUMrShVkMBghA0cBAkoXMAW6emspnCkoxUyxiUISmw2DAqGwNHkmEq6gKfEEojp70+OuqXgk1l8zj81kRJi6iIF5nMlsCej0Dg9nQ6BgkATl00VBGVqvVmu131YACURNdeIco7wjXIGabmYLLSZrTC7Q6HAKYZbUaKcwYHH7DAGFbNiIpaPIAGY0DJ4eSkTZcfXFZMfVPfM0IRFGAg2Bxg+qheo9XoeRDO12cj1err+P1VhI1sDqLJQZg0NAaMD5OMJpOlY19pmgGoRBwEOHi+HzsFOl0jxeeror33+yaBmsQPgsA0JgKCwLAepFIa54phU-bpnoEiGMKSKelYX7OK+C7up+3qrr+mIbkkEBwGATCQQaJRvLBjI-Ne+jWAYjR2OW4p+pOoTWAEWHvjhy4+muf7VsR9AsBR3Ywb2cFXjo+jGGYlg2PYjjOG4s6Dm+bpLl+AkEfKRGEA2KDKpAmpLCs6ybDGJ6Jj2JrwfRtQToCwLtN64LqfOvHaXhP4xJM24kfA7z-vg9LSXRskIAAtNx6mxcKylJcpJjrtihAkOQlDUHQjAsOFtEDvCoR3iYAQSP4nElSYqICoM5ZGAELh+OYfoGOYaVBplqQ5Rk2S5P0NFpo5bLOPeXGIjYRgSAYfh1UhBANU1PQIm15gGJ1Na4hZBJgAVw1RaydgEJOARNQYs3dA45jzXejGMWOkR+mKm1JCGaoUOGB32TJNQugKJUneK1hZpyATWCVnGvRldaak2LZtntF4RQOoIjlxFpZnUZXikYPHBJ65UBNdnJcUY0MEFuO57ge6hHvtDlRWx95QrCJgzQ4Z0dPjD5+iTkobUJBkEIBVAgWBwVDYzNQleY96E9NEjBMr-KeW+BNsnzYoCxTQVkaQDO-fohg+MiFqSlC5itO1POE1rpOC4R6Ui6JzCG5FfwuBII4NP4rWwi0DjWLbmvE9rZMU0ZJkQGZeKWe7RVBHLAeTVKM1zWrma82HDsUxkaA4GA7Yx9t+IbAn6a2rNwoRBVQeKWK83MTNyKzS6Y7BKE-lREAA */
```
Comment (spyapali) [inline phaser-react/src/gameplay.js:4]: What does this represent?

File: phaser-react/src/App.jsx
Comment (spyapali) [inline phaser-react/src/App.jsx:1]: I really like how you've organized this file! Kudos! 
Especially with the object where you map `xstate state value` to component.

File: phaser-react/src/gameplay.js [hunk 2]
```diff
@@ -29,21 +30,14 @@ export const gameplayMachine = createMachine({
...
-                },
-            },
-        },
-        decision_AB: {
+        objective: {
             on: {
                 ILLEGAL_OPTION_SELECTED: {
-                    target: "decision_A1",
+                    target: "find_coyote",
                     // cond: "isSelectedIllegalOption",
```
Comment (spyapali) [inline phaser-react/src/gameplay.js:41]: Would you want to add a TODO here on top of the commented sections of code to indicate to the reviewer that you'd want to revisit adding this in (`cond`, etc.)?

**Review by spyapali — APPROVED** [review]
LGTM! 🎉

also: 
> added xstate visualizer extension

kudos to this :) I just did so too and it looks to be really useful!

**Comment by spyapali** [comment]
> For catalyst_earth and catalyst_country, I render these states conditionally inside the Catalyst component, whereras for other states that also only have a continue button, I render the states in their own seperate components. Would best practice be to group similar scenes like Catalysts scene or create seperate components for every state? 

Good question! Putting them in different components makes a bit more sense to me semantically because they are different screens and are mapped to different states. That being said there's no hard, fast rule on what's right!

**Comment by spyapali** [comment]
> I'm also getting error " {send} is missing in props validation" so i turned this rule off in eslintrc file.

This makes sense, but adding proptypes can be helpful for validating that the props passed into a component are of the correct type. You can add it in later. 

Here's an example of how this would look like using javascript (i pulled this from Chat GPT): 

```javascript
import React from 'react';
import PropTypes from 'prop-types';

const MyComponent = ({ send }) => {
  // component implementation
};

MyComponent.propTypes = {
  send: PropTypes.func.isRequired, // or appropriate type
};
```

File: phaser-react/src/gameplay.js (hunk 1 above)
Comment (mtp000) [reply phaser-react/src/gameplay.js:4]: it's the xstate visualizer! every time it's used, it creates this comment. not sure why.

File: phaser-react/src/gameplay.js (hunk 2 above)
Comment (mtp000) [reply phaser-react/src/gameplay.js:41]: would the guards be necessary here to control the state transition if for this state's component file, i just created 2 buttons that takes the player to the appropriate next state?

Comment (spyapali) [reply phaser-react/src/gameplay.js:41]: hmm good question - I'm not entirely sure! but worth it for you to try!

**Pull request merged**